tests/rhui3_tests/test_CLI.py

```diff
@@ -0,0 +1,112 @@
+#! /usr/bin/python -tt
+
+import nose, stitches, logging, yaml
+
+from rhui3_tests_lib.rhuimanager import *
+from rhui3_tests_lib.rhuimanager_repo import *
+from rhui3_tests_lib.rhuimanagercli import *
+
+from os.path import basename
+
+logging.basicConfig(level=logging.DEBUG)
+
+connection=stitches.connection.Connection("rhua.example.com", "root", "/root/.ssh/id_rsa_test")
+
+with open('/tmp/rhui3-tests/tests/rhui3_tests/tested_repos.yaml', 'r') as file:
+    doc = yaml.load(file)
+
+yum_repo_name_1 = doc['CLI_repo1']['name']
+yum_repo_id_1 = doc['CLI_repo1']['id']
+yum_repo_name_2 = doc['CLI_repo2']['name']
+yum_repo_id_2 = doc['CLI_repo2']['id']
+
+custom_repo_name = "my_custom_repo"
+
+def setUp():
+    print "*** Running %s: *** " % basename(__file__)
+
+def test_01_initial_run():
+    '''Do an initial rhui-manager run to make sure we are logged in'''
+    RHUIManager.initial_run(connection)
+
+def test_02_check_empty_repo_list():
+    '''Check if the repolist is empty (interactively; not currently supported by the CLI)'''
+    nose.tools.assert_equal(RHUIManagerRepo.list(connection), [])
+
+def test_03_remove_existing_entitlement_certificates():
+    '''Clean up uploaded entitlement certificates'''
+    RHUIManager.remove_rh_certs(connection)
+
+def test_04_create_custom_repo():
+    '''Create a custom repo for further testing (interactively; not currently supported by the CLI)'''
+    RHUIManagerRepo.add_custom_repo(connection, custom_repo_name, entitlement="n")
+
+def test_05_upload_entitlement_certificate():
+    '''Upload the Atomic (the small) entitlement certificate'''
+    RHUIManagerCLI.cert_upload(connection, "/tmp/extra_rhui_files/rhcert_atomic.pem", "Atomic")
+
+def test_06_check_certificate_info():
+    '''Check certificate info for validity'''
+    RHUIManagerCLI.cert_info(connection)
+
+def test_07_check_certificate_expiration():
+    '''Check if the certificate expiration date is OK'''
+    RHUIManagerCLI.cert_expiration(connection)
+
+def test_08_check_unused_product():
+    '''Check if a repo is available'''
+    RHUIManagerCLI.repo_unused(connection, yum_repo_name_1)
+
+def test_09_add_rh_repo_by_product():
+    '''Add a Red Hat repo by its product name'''
+    RHUIManagerCLI.repo_add(connection, yum_repo_name_1)
+
+def test_10_add_rh_repo_by_id():
+    '''Add a Red Hat repo by its ID'''
+    RHUIManagerCLI.repo_add_by_repo(connection, [yum_repo_id_2])
+
+def test_11_repo_list():
+    '''Check the added repos'''
+    RHUIManagerCLI.repo_list(connection, yum_repo_id_1, yum_repo_name_1)
+    RHUIManagerCLI.repo_list(connection, yum_repo_id_2, yum_repo_name_2)
```

Comment:
@RadekBiba questions here:
1) can it be changed to except not one repo, but a list of repos? 
2) do we need to check that no any other repos were added?
3) if 1 and 2, can it be changed to `list_of_repos=list of repos`, and not `one_item is in the list`? 

[1] https://github.com/RedHatQE/rhui3-automation/blob/master/tests/rhui3_tests_lib/rhuimanager_repo.py#L148
[2] https://github.com/RedHatQE/rhui3-automation/blob/master/tests/rhui3_tests/test_repo_managment.py#L39

Reply:
1. there are two arguments: 1 - repo ID to add, 2 - repo name to check that the expected repo was actually added. So the change would require two lists, a list of repo IDs and a list of repo names, with the same order. And then the checking would have to go through the lists and combine the IDs and names and check the `rhui-manager repo list` output len(list) times. From the complexity point of view, I see no benefit here (no real optimization), while the current approach is clear and simple from the function caller's point of view.
2. I don't see how that could happen, but I think it's a good idea. I'll add that test.
3. As I wrote in 1, there would have to be two lists to properly check if the right repos were added. I'd say the current approach + item 2 will cover this, though.